src/main/java/fr/sncf/osrd/infra_state/SwitchPost.java

```diff
@@ -0,0 +1,196 @@
+package fr.sncf.osrd.infra_state;
+
+import fr.sncf.osrd.infra.SuccessionTable;
+import fr.sncf.osrd.infra.TVDSection;
+import fr.sncf.osrd.infra.routegraph.Route;
+import fr.sncf.osrd.train.Train;
+import fr.sncf.osrd.train.TrainStatus;
+import fr.sncf.osrd.simulation.Simulation;
+import fr.sncf.osrd.simulation.SimulationError;
+import fr.sncf.osrd.infra.Infra;
+
+import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class SwitchPost {
+
+    private HashMap<String, SuccessionTable> tables;
+    private HashMap<String, Integer> currentIndex;
+    private HashMap<String, HashMap<String, Integer>> occurences;
+    private HashMap<String, HashSet<Request>> waitingList;
+    private HashMap<String, String> lastRequestedRoute;
+    private HashMap<String, String> currentTrainAllowed;
+
+    public SwitchPost() {
+        tables = null;
+    }
```

Comment:
It would be simpler and safer if we manage to initialize everything in the constructor. If we really need the initialized infra to init everything, then this init call should be at the end of the Infra constructor. It may be tricky to initialize everything in the right order.